packages/vite-plugin-nitro/src/lib/vite-plugin-nitro.ts

```diff
@@ -368,3 +372,16 @@ const withCloudflareOutput = (nitroConfig: NitroConfig | undefined) => ({
     serverDir: '{{ output.publicDir }}/_worker.js',
   },
 });
+
+const isNetlifyPreset = (buildPreset: string | undefined) =>
+  process.env['NETLIFY'] ||
+  process.env['NETLIFY_LOCAL'] ||
+  (buildPreset && buildPreset.toLowerCase().includes('netlify'));
+
+const withNetlifyOutput = (nitroConfig: NitroConfig | undefined) => ({
+  ...nitroConfig,
+  output: {
+    ...nitroConfig?.output,
+    dir: '{{ rootDir }}/.netlify/functions-internal',
```

Comment:
I think what we need to do here is set the `serverDir`. The `dir` sets the output for everything else, including the `publicDir` and `.nitro` build directory.

```suggestion
    serverDir: '{{ rootDir }}/.netlify/functions-internal',
```